File: app/src/main/java/com/boardgamegeek/util/DataUtils.java
package com.boardgamegeek.util;

import android.text.TextUtils;

public class DataUtils {
	public static String generatePollResultsKey(int level, String value) {
		if (level <= 0) {
			return generatePollResultsKey("", value);
		} else {
			return generatePollResultsKey(String.valueOf(level), value);
		}
	}

	public static String generatePollResultsKey(String level, String value) {
		String key = level;
		if (TextUtils.isEmpty(key)) {
			key = value;
			int index = key.indexOf(" ");
			if (index > -1) {
				key = key.substring(0, index);
			}
		}
		return key;
	}
}

File: app/src/main/java/com/boardgamegeek/util/DataUtils.kt
package com.boardgamegeek.util

object DataUtils {
    fun generatePollResultsKey(level: Int, value: String): String {
        return if (level <= 0) {
            generatePollResultsKey("", value)
        } else {
            generatePollResultsKey(level.toString(), value)
        }
    }

    fun generatePollResultsKey(level: String?, value: String): String {
        return if (level.isNullOrBlank()) {
            value.substringBefore(" ")
        } else {
            level
        }
    }
}
